pkg/osvscanner/osvscanner.go

```diff
@@ -103,7 +106,7 @@ const (
 //   - Any lockfiles with scanLockfile
 //   - Any SBOM files with scanSBOMFile
 //   - Any git repositories with scanGit
-func scanDir(r reporter.Reporter, dir string, skipGit bool, recursive bool, useGitIgnore bool, compareOffline bool) ([]scannedPackage, error) {
+func scanDir(r reporter.Reporter, dir string, skipGit bool, recursive bool, useGitIgnore bool, compareLocally bool) ([]scannedPackage, error) {
```

Comment:
I don't think this new name is accurate - comparing offline is a very different concern from comparing locally; while that different doesn't matter too much in this function, I think for the sake of consistency it's better to keep this as `compareOffline`

Reply:
could you explain more about the difference?

I had a discussion with @another-rex and we agree that both indicates users not willing to send package information to API (which is required by dependency resolution). 

if `CompareOffline` is true, `CompareLocally` is also true, but not vice versa.

If we check `CompareLocally`, and it is false, both `CompareOffline` and `CompareLocally` are false. 
If we only check `CompareOffline`, when `CompareOffline` is false there might be chance that `CompareLocally` still true, but package information will still be sent to deps.dev API and I think that is not what a user wants.

I have to say I also felt confused, so that's why I use `allowResolver` first and check if both `CompareOffline` and `CompareLocally` false.

Reply:
+1 to Xueqin's comment, that's also my understanding. I believe the only usage of compareLocally here in scanDir is for the vendored directories scanning, which should be turned off when compareLocally is on for 2 reasons:

- It sends granular information (hashes of all files)
- The results are very likely to be affected git commits, which we don't scan in local mode anyway, so there's no point to the scan.

Reply:
I don't disagree with the thinking, but imo it doesn't match with the name - really we should change it to be "compare offline" and "download offline databases", because that's why we have the two flags, and that'd resolve this confusion